pkg/cli/admin/mustgather/summary.go

```diff
@@ -0,0 +1,179 @@
+		missingOperators.Delete(clusterOperator.Name)
+		clusterOperatorSummary := humanSummaryForClusterOperator(clusterOperator)
+		if len(clusterOperatorSummary) == 0 { // not noteworthy
+			continue
+		}
+		clusterOperatorStrings = append(clusterOperatorStrings, humanSummaryForClusterOperator(clusterOperator))
+	}
+
+	for _, missingOperator := range missingOperators.List() {
+		clusterOperatorStrings = append(clusterOperatorStrings, fmt.Sprintf("clusteroperator/%s is missing", missingOperator))
+	}
+
+	if len(clusterOperatorStrings) > 0 {
+		return "\t" + strings.Join(clusterOperatorStrings, "\n\t")
+	}
+	return "\tAll healthy and stable"
+}
+
+func humanSummaryForClusterOperator(clusterOperator configv1.ClusterOperator) string {
+	available := v1helpers.IsStatusConditionTrue(clusterOperator.Status.Conditions, configv1.OperatorAvailable)
+	progressing := !v1helpers.IsStatusConditionFalse(clusterOperator.Status.Conditions, configv1.OperatorProgressing)
+	degraded := !v1helpers.IsStatusConditionFalse(clusterOperator.Status.Conditions, configv1.OperatorDegraded)
+	upgradeable := !v1helpers.IsStatusConditionFalse(clusterOperator.Status.Conditions, configv1.OperatorUpgradeable)
+
+	availableMessage := "<missing>"
+	progressingMessage := "<missing>"
+	degradedMessage := "<missing>"
+	upgradeableMessage := "<missing>"
+	if condition := v1helpers.FindStatusCondition(clusterOperator.Status.Conditions, configv1.OperatorAvailable); condition != nil {
+		availableMessage = condition.Message
+	}
+	if condition := v1helpers.FindStatusCondition(clusterOperator.Status.Conditions, configv1.OperatorProgressing); condition != nil {
+		progressingMessage = condition.Message
+	}
+	if condition := v1helpers.FindStatusCondition(clusterOperator.Status.Conditions, configv1.OperatorDegraded); condition != nil {
+		degradedMessage = condition.Message
+	}
+	if condition := v1helpers.FindStatusCondition(clusterOperator.Status.Conditions, configv1.OperatorUpgradeable); condition != nil {
+		upgradeableMessage = condition.Message
+	}
+
+	switch {
+	case !available:
+		return fmt.Sprintf("clusteroperator/%s is not available (%v) because %v", clusterOperator.Name, availableMessage, degradedMessage)
+	case degraded:
+		return fmt.Sprintf("clusteroperator/%s is degraded because %v", clusterOperator.Name, degradedMessage)
+	case !upgradeable:
+		return fmt.Sprintf("clusteroperator/%s is not upgradeable because %v", clusterOperator.Name, upgradeableMessage)
+	case progressing:
+		return fmt.Sprintf("clusteroperator/%s is progressing: %v", clusterOperator.Name, progressingMessage)
+	case available && !progressing && !degraded && upgradeable:
+		return ""
+	default:
+		return fmt.Sprintf("clusteroperator/%s is in an edge case", clusterOperator.Name)
+	}
+}
+
+func humanSummaryForClusterVersion(clusterVersion *configv1.ClusterVersion) string {
+	if clusterVersion == nil {
+		return "Cluster is in a version state we don't recognize"
```

Comment:
Can we make this message more detailed? 
If we hit this this seems 'bad' and we likely need to take some type of operation to get this cluster healthy. 

@sdodson  can the install team help in this space to outline what this error means and why one might hit it? As well as what one should look at in the must-gather to debug/diagnose this. 

IE: General catchall's like this warrant a 'this is known bad' situation to be in, and should come with some 'known next steps', like a link to a KCS?

Reply:
> Can we make this message more detailed?

No.  We don't know what this is.  You have the entire clusterversion object for a human/machine to look at.